### PR TITLE
fix some typos in api/types

### LINKED ISSUE #53420: Got some typos in api/types
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
Some typos in api/types

**What you expected to happen**:
Should fix them

**How to reproduce it (as minimally and precisely as possible)**:

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
Fix some typos in api/types

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #53420

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by ericchiang** [comment]
These fixes need to be applied to the external versioned APIs too: https://github.com/dixudx/kubernetes/blob/07c2010947d1354b1a917f10d416549e76802881/staging/src/k8s.io/api/core/v1/types.go#L4403

Also, running https://github.com/client9/misspell on that file indicates another spelling error.

```
$ wget https://raw.githubusercontent.com/dixudx/kubernetes/07c2010947d1354b1a917f10d416549e76802881/pkg/api/types.go
$ misspell types.go 
types.go:1126:28: "mulitple" is a misspelling of "multiple"
```

Mind updating the versioned API and running a spell checker against other types.go files as part of this PR?

File: pkg/api/types.go [hunk 1]
```diff
@@ -2919,7 +2919,7 @@ type NodeSpec struct {
 type DaemonEndpoint struct {
 	/*
 		The port tag was not properly in quotes in earlier releases, so it must be
-		uppercased for backwards compat (since it was falling back to var name of
+		uppercase for backwards compatible (since it was falling back to var name of
```
Comment (roberthbailey) [inline pkg/api/types.go]: compatibility

**Comment by roberthbailey** [comment]
ok to test

**Comment by dixudx** [comment]
@roberthbailey PTAL. Thanks.

**Comment by dixudx** [comment]
@lavalamp @brendandburns @smarterclayton @colhom Needs your `/approve no-issue`. Thanks so much.

**Comment by dixudx** [comment]
@lavalamp @brendandburns @smarterclayton @colhom Needs your /approve no-issue. Thanks so much.

File: pkg/apis/certificates/types.go [hunk 2]
```diff
@@ -122,7 +122,7 @@ type KeyUsage string
 const (
 	UsageSigning            KeyUsage = "signing"
 	UsageDigitalSignature   KeyUsage = "digital signature"
-	UsageContentCommittment KeyUsage = "content committment"
+	UsageContentCommittment KeyUsage = "content commitment"
```
Comment (jbeda) [inline pkg/apis/certificates/types.go]: This is more than just a comment and may have compatibility impact.  Note also that the const is misspelled also.

This also lines up with the cloudflare library that this string gets passed to: https://sourcegraph.com/github.com/kubernetes/kubernetes@886e04f1fffbb04faf8a9f9ee141143b2684ae68/-/blob/vendor/github.com/cloudflare/cfssl/config/config.go#L460-461.  I'm guessing that we don't want to change this one.

**Review by jbeda — CHANGES_REQUESTED** [review]
We shouldn't be changing actual constants in this PR.

**Comment by jbeda** [comment]
@mikedanese Looks like you added `contentCommittment`.  Want to weigh in here?

File: pkg/apis/certificates/types.go (hunk 2 above)
Comment (dixudx) [inline pkg/apis/certificates/types.go]: @jbeda Updated. PTAL. Thanks.

**Comment by dixudx** [comment]
@ericchiang @jbeda @roberthbailey Updated. PTAL. Thanks.

**Comment by dixudx** [comment]
ping @ericchiang @jbeda @roberthbailey Rebased. PTAL. Thanks.

**Review by jbeda — APPROVED** [review]

**Comment by jbeda** [comment]
Thanks!

**Comment by roberthbailey** [comment]
@dixudx - thanks for the fixes and sorry it was such an arduous process to get them merged.

**Pull request merged**